### PR TITLE
1)Updating the names of methods used in app to app authentication

### PR DESCRIPTION
**Pull request by sowmiya**
2)Introduced an instance method to handle authenticating users from
a launch URL

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sowmiya has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h [hunk 1]
```diff
@@ -31,20 +31,28 @@
 - (void)authenticateInAppWithCompletionBlock:(void (^)(BOXUser *user, NSError *error))completion;
 
 /**
- *  Discerns whether a launch URL is associated with Box authentication - if it is the return URL that should be used
+ *  Discerns whether a launch URL is associated with Box app to app authentication - if it is the return URL that should be used
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h]: minor: styling, App-to-App?

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h [hunk 2]
```diff
@@ -31,20 +31,28 @@
...
  *  to complete the user's authentication in the case of authenticating App-to-App using the Box app.
  *
  *  @param authenticationURL    The URL with which the app was launched.
  */
-+ (BOOL)canCompleteAuthenticationWithURL:(NSURL *)authenticationURL;
++ (BOOL)canCompleteAppToAppAuthenticationWithURL:(NSURL *)authenticationURL;
 
 /**
- *  Complete the user's authentication to the account that is in use in the Box app.
+ *  Complete the user's authentication to the account that is in use in the Box app.(App-to-App)
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h]: Minor: Missing space.

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h [hunk 3]
```diff
@@ -31,20 +31,28 @@
...
-+ (void)completeAuthenticationWithURL:(NSURL *)authenticationURL;
++ (void)completeAppToAppAuthenticationWithURL:(NSURL *)authenticationURL;
+
+/**
+ *  Complete the user's authentication from the URL with which the app was launched.
+ *
+ *  @param authenticationURL    The URL with which the app was launched.
+ *  @param completionBlock      Called when the authentication has completed.
+ */
+- (void)completeAuthenticationWithURL:(NSURL *)authenticationURL completionBlock:(void (^)(BOXUser *user, NSError *error))completion;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h]: Should there be another boolean helper method to be able to tell if this method can/should be used? To differentiate from the case where neither one should be used.

**Comment by boxtcarpel** [comment]
+2
Just one question (and we should look at the Travis failure).

:pear:

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h (hunk 1 above)
Comment (sowmiya) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h]: done.

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h (hunk 2 above)
Comment (sowmiya) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h]: done

File: BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h (hunk 3 above)
Comment (sowmiya) [inline BoxContentSDK/BoxContentSDK/Clients/BOXContentClient+Authentication.h]: We discussed offline to not expose this publicly. I have moved it to a private header.

**Comment by sowmiya** [comment]
I ran the tests and they succeeded. I believe Travis is failing due to timing issue.

**Comment by boxtcarpel** [comment]
+2

:tanabata_tree:

**Pull request merged**